modules/extensions-support/src/test/java/org/mule/runtime/module/extension/internal/runtime/operation/ComponentMessageProcessorTestCase.java

```diff
@@ -227,11 +246,16 @@ public void multipleUpstreamPublishers() throws MuleException, InterruptedExcept
         .subscribe(eventsConsumer2);
 
     eventsEmitter.start();
-    eventsEmitter2.start();
-    eventsConsumer.await();
-    eventsEmitter.stop();
-    eventsConsumer2.await();
-    eventsEmitter2.stop();
+    try {
+      eventsEmitter2.start();
+      eventsConsumer.await();
+      eventsEmitter.stop();
+      eventsConsumer2.await();
+      eventsEmitter2.stop();
+    } catch (Throwable t) {
+      eventsEmitter.stop();
+      eventsEmitter2.stop();
```

Comment:
could we just do these `stop`s in a `finally` block?

Reply:
Yes, I think we could. I was trying to make it messy by completing the first publisher before awaiting for the second subscriber to have received all items. But in reality it makes no sense, because it could have received all the events already anyway, because I already did both starts.
So yes, let's simplify it.

Reply:
Done